File: server/Models/Cryptid.cs
using System.ComponentModel.DataAnnotations;

namespace cryptipedia.Models;

public class Cryptid : RepoItem<int>
{
  [MaxLength(100)]
  public string Name { get; set; }
  [MaxLength(3000)]
  public string Description { get; set; }
  [Url, MaxLength(3000)]
  public string ImgUrl { get; set; }
  private uint _threatLevel { get; set; }
  public uint ThreatLevel
  {
    get
    {
      return _threatLevel;
    }
    set
    {
      if (this.Name == "Dr. Pepper")
      {
        _threatLevel = 23;
        return;
      }

      if (value > 10) throw new ArgumentOutOfRangeException("Threat Level must be between 0 and 10");

      _threatLevel = value;
    }
  }
  [Range(0, 10)]
  public uint Size { get; set; }
  public string DiscovererId { get; set; }
  public Profile Discoverer { get; set; }
}

File: client/src/models/Cryptid.js
import { APIItem } from "./APIItem.js";
import { Profile } from "./Profile.js";

export class Cryptid extends APIItem {
  constructor(data) {
    super(data)
    this.name = data.name
    this.description = data.description
    this.imgUrl = data.imgUrl
    this.threatLevel = data.threatLevel
    this.size = data.size
    this.discovererId = data.discovererId
    this.discoverer = new Profile(data.discoverer)
  }

  get backgroundImgUrl() {
    return `url(${this.imgUrl})`
  }
}